### PR TITLE
Respect APPLY_DEFAULT_VALUES

### PR DESCRIPTION
**Pull request by nick-benoit14**
Adds support for proposed APPLY_DEFAULT_VALUES deserialization feature

### CODE REVIEW

File: src/main/scala/com/fasterxml/jackson/module/scala/introspect/ScalaAnnotationIntrospectorModule.scala [hunk 1]
```diff
@@ -100,14 +100,15 @@ object ScalaAnnotationIntrospector extends NopAnnotationIntrospector with ValueI
     extends StdValueInstantiator(delegate) {
 
     private val overriddenConstructorArguments: Array[SettableBeanProperty] = {
+      val applyDefaultValues = config.hasDeserializationFeatures(DeserializationFeature.APPLY_DEFAULT_VALUES.getMask)
```
Comment (pjfanning) [inline src/main/scala/com/fasterxml/jackson/module/scala/introspect/ScalaAnnotationIntrospectorModule.scala:103]: this looks wrong - code that checks if a feature is enabled generally looks like this

```
  override def mapArray(jp: JsonParser, ctxt: DeserializationContext): AnyRef = {
    if (ctxt.isEnabled(DeserializationFeature.USE_JAVA_ARRAY_FOR_JSON_ARRAY)) {
```

Comment (pjfanning) [reply src/main/scala/com/fasterxml/jackson/module/scala/introspect/ScalaAnnotationIntrospectorModule.scala:103]: better still use `config: DeserializationConfig` -- it is available where your code change is -- config.isEnabled(...)

Comment (nick-benoit14) [reply src/main/scala/com/fasterxml/jackson/module/scala/introspect/ScalaAnnotationIntrospectorModule.scala]: Switched to `ctxt.isEnabled(DeserializationFeature.APPLY_DEFAULT_VALUES)` I had to move where I did the check to the within the null value provider in order to make it so ObjectReaders could be configured separately.

**Comment by nick-benoit14** [comment]
@pjfanning  @cowtowncoder  Any more thoughts on this? I also have a  corresponding PR on jackson-databind [here](https://github.com/FasterXML/jackson-databind/pull/3193) as well

**Pull request merged**